query-result.hh

```diff
@@ -426,6 +433,14 @@ public:
 
     void ensure_counts();
 
+    const std::optional<full_position> last_position() const {
```

Comment:
why not return a `const std::optional<full_position>&` and let the caller copy it if needed?
also, it makes it trivially noexcept :)

Reply:
I think that was my intention, just forgot the `&`.

Reply:
I don't think `noexcept` adds anything to this API though.